### ISSUE TITLE
dr_wav: Support for Unicode filenames

### ISSUE THREAD

**Issue by SephiRok** [mackron/dr_libs#61]
wchar_t* and _wfopen() is required for Unicode filename support on Windows.

See https://en.cppreference.com/w/cpp/filesystem/path/u8path for an example on how to handle it.

If the API cannot be adjusted for Unicode filenames, I would suggest adding function to the API that accept FILE* so it can be opened correctly before.

**Comment by mackron** [comment]
I have this planned for all my libraries. Related issue in miniaudio: https://github.com/dr-soft/miniaudio/issues/68

I'll add a drwav_init_file_w() for wchar_t, and perhaps also explicit UTF-8/16/32 versions as well. Will get onto this as soon as I get a chance.

**Comment by mackron** [comment]
By the way, I forgot to mention this earlier, if you're in need of this right now you can really easily open the `drwav` object using the callbacks API. The callbacks map cleanly with fread() and fseek(). Example:

```c
size_t on_read_FILE(void* pUserData, void* pBufferOut, size_t bytesToRead)
{
    return fread(pBufferOut, 1, bytesToRead, (FILE*)pUserData);
}

drwav_bool32 on_seek_FILE(void* pUserData, int offset, drwav_seek_origin origin)
{
    return fseek((FILE*)pUserData, offset, (origin == drwav_seek_origin_current) ? SEEK_CUR : SEEK_SET) == 0;
}

...

pMyFILE = _wfopen(...);

drwav_init(&myWav, on_read_FILE, on_seek_FILE, pMyFILE);
```

**Comment by mackron** [comment]
I've added support for this to the "wav/dev" branch. You can see the new APIs in the commit message: https://github.com/mackron/dr_libs/commit/c93953883a0b3c68f80ad2b107a2f5e7a7c932dd

Does this satisfy your requirements?

**Comment by mackron** [comment]
Merged into master in version 0.10.0. Feel free to reopen this issue if you encounter an problems.
